### ISSUE TITLE
Automatic shutdown after inactivity

### ISSUE THREAD

**Issue by Eduardoesteban1694** [FrSkyRC/ETHOS-Feedback-Community#4100]
Is there an option for the transmitter to turn off automatically after a certain amount of inactivity?

**Comment by BladeScraper-Designs** [comment]
+1 for this... I think most radios have it by now.

**Comment by bsongis-frsky** [comment]
How can we introduce this feature and stay safe?

**Comment by azaz44** [comment]
> How can we introduce this feature and stay safe?

only shutdown if:
- telemetry not there (rx disconnected or has no telemetry)
- throttle cut is engaged (system value from built-in thr mix)

**Comment by RealTadango** [comment]
Always make it a opt-in option. I don't want this ever.

**Comment by spoke2570** [comment]
Ya, definitely an option to turn it off. Hour plus autonomous missions would worry me if I thought I could have an accidental shutdown.

**Comment by azaz44** [comment]
Yes, for sure this needs a setting and should be off be default (doesn't shutdown by default).
And I think having ability to disable this function per model might be useful too.

**Comment by raydars** [comment]
100% don't want this as a default setting...    if it must be then i tmust be optional and  it must never operate without receiver failsafes set.

**Comment by jasalca** [comment]
The option of shutting down is risky. I would prefer it to beep sounds at the user's will

**Comment by MacielRMelo** [comment]
The mere fact that this option exists is an unnecessary risk, even though it is a deactivated default option. The beep is more than enough for the user to turn off the transmitter.
Better a dead transmitter battery than the radio turning off while the plane is in flight or an electric plane is active.
I would like to not see this feature implemented.

**Comment by dehigama** [comment]
"How can we introduce this feature and stay safe?"

in addition to the above specially the 'check' to see if the Tx is still communicating with an Rx. I would say a visual count down timer that you can close clicking a button on the screen or any movement of sticks /sliders, if equipped the internal gyro.

So basically only after passing through multiple AND conditions to activate it. and many ways to deactivate it.

**Comment by azaz44** [comment]
> "How can we introduce this feature and stay safe?"
> 
> in addition to the above specially the 'check' to see if the Tx is still communicating with an Rx. I would say a visual count down timer that you can close clicking a button on the screen or any movement of sticks /sliders, if equipped the internal gyro.
> 
> So basically only after passing through multiple AND conditions to activate it. and many ways to deactivate it.

Shutdown would only happen in case of long inactivity, which means you were not touching your sticks, switches or pots for last 10 mins or so. And you would have "long inactivity" warnings long before shutdown. So I think there's no need for countdown you mentioned.

In my opinion this can be made very safe:
- only shutdown after long inactivity
- only shutdown if warnings were issued for a long time (so shutdown time is longer than inactivity time)
- only shutdown if telemetry not transmitting
- only shutdown if in throttle cut
- could also add, only shutdown if internal gyro didn't detect movements (but this has a downside, it will not shutdown when you're driving and your radio is in your car)

It would never shutdown when flying any normal RC models.

I think the only concerning situation is autonomous flight, where the model is in air for a very long time, flying or hovering on its own, user doesn't touch radio, ignores inactivity warnings (left to the restroom), and has enabled this function in the first place. Plus has RX with no telemetry or has lost telemetry for 10 minutes, and doesn't use built-in throttle mix with throttle cut function. I think 100% safety in such case is not possible.

**Comment by RealTadango** [comment]
Also, only shutdown if telemetry not transmitting BUT has received telemetry during the session (not all models use telemetry). This also must work correctly with MPM which keeps sending data....

**Comment by jasalca** [comment]
I wonder if it's so hard to turn off the radio with your finger.

**Comment by dehigama** [comment]
"I wonder if it's so hard to turn off the radio with your finger."

its more about forgetting to turn it off.

**Comment by jasalca** [comment]
That's exactly what an audible warning is for

**Comment by dehigama** [comment]
"That's exactly what an audible warning is for"

I agree, when you are near to it, but usually we fly in a club with all sorts of noises. members chatting, models in the air, generators powering up the charging bay (loads of beeps).

**Comment by BladeScraper-Designs** [comment]
Yeah I don't see what the big deal is.  This is a feature that has been done plenty of times in other systems.  It's not hard to say look, no inputs at all for 10 minutes, and no incoming telemetry?  Give a warning.  No more for another 5 minutes, still no telemetry and no cancellation?  Shut down.  

Obviously it wouldn't work in all situations, hence it being optional to enable or disable it globally, or ideally per model.  But it wouldn't be the first radio to implement such a feature, and it shouldn't be hard to do.

Another option is to put the radio in a low power state, kind of a sleep mode.  Put the MCU into a low power state, shut down all but the most low level functionality, shut down RF but keep a wakeup timer that periodically wakes up RF and checks for incoming telemetry.  For example if a receiver is plugged in/powered on, wake up the radio and return it to normal ops.  It could even detect the receiver that's been powered on and automatically switch to the corresponding model automatically. This is precisely how Vbar Control behaves, and Spektrum has a similar method on their radios.  It could be very cool for Ethos.

Maybe you're perfect, and never ever forget to turn your radio off and always hear it when it says "prolonged inactivity", but I'm sure not.  The speaker may be loud enough to hear when you're standing right beside it or holding it, but when I set my radio down after flying, then get distracted by someone asking for my help and I walk over to the other side of the field, it's maybe not quite loud enough.

**Comment by jasalca** [comment]
It is clear that behind all this there is a team of professionals who will value it. I'm just stating my opinion, because just as you can leave the radio on, you can also leave the plane on and that's not going to turn off by itself.

Speaking of slip-ups on my airfield, a colleague forgot to turn off the equipment and when the radio turned off automatically, the plane taxied and injured another colleague who was flying.

He probably had the fail safe configured wrong

**Comment by BladeScraper-Designs** [comment]
That is why "no telemetry active" is a given condition for automatic shutdown.  I don't think that should even be an option... Telemetry incoming?  Never shut down.

I understand your concerns and agree it needs to be implemented carefully and explained explicitly in the manual situations where you should not enable it, but I do think it's a feature worth having.  Personally it's something I've wanted ever since I started using Ethos.

**Comment by aeroplanebelg** [comment]
l'arrêt automatique de l'émetteur existe chez futaba depuis la sortie de la 9z (30 ans) . cet arrêt  est réglable de 10 à 60 minutes.
Pourquoi ? Pour permettre l ' écolage sans risque. Si ça posait problème, ça se saurait depuis longtemps !!!! Moi je suis pour à 100 %. Pour rappel l'arrêt n'est effectif que si on ne touche pas un manche pendent le temps programmé !!!! Une heure sans toucher un manche?? HUM c'est tout simplement un oubli . ça reste une option pour les grands distraits mais utile. Bons vols à tous.

**Comment by MacielRMelo** [comment]
Distraction is not welcome for those who fly model airplanes. In several countries, fgovernment flight control agencies already legislate on model aircraft. 
A model airplane is not a toy!
Having a function that can simply turn off the radio seems very "delicate".
Anyone who works with systems knows that there is something they call a "bug". Even if you think about all the conditions that were placed here (do not touch the radio, gyroscope, telemetry, stick, etc, etc, etc, etc), if there is a "bug", for whatever reason, the radio will be "OFF " in flight!!!!
Why implement such a delicate function, just to serve half a dozen (or a few more!) distracted pilots?
I've been flying for 25 years and I've only left the radio "on" unintentionally a few times, in fact, twice: 1) one time, I replaced the transmitter's battery with the battery from a friend's radio and flew normally for the rest of the day; 2) another time, I came home! 
I ask developers to carefully evaluate the cost-benefit of this feature. Even though it is a feature that can be turned off, the simple fact that it exists in the code is a problem.
Just remember the time and number of versions to "stabilize" the conditions for the radio to enter emergency mode. Just remember that.
Sorry to those who don't agree, but I needed to express my opinion.

**Comment by jasalca** [comment]
Totally agree with you, it couldn't be better explained.

**Comment by Dsrenger** [comment]
Thanks Maciel,
Totally agree...

Stay focused on what you are doing. You are flying and handling a RC model...

**Comment by FlipAirboost** [comment]
> How can we introduce this feature and stay safe?

The TX could perhaps do a check when you turn it on to check that you have a brain and memory!

- Brain test: A labyrinth is displayed on the screen when the transmitter is set up, a tank must cross it and find the output piloted with the two gimbals.

Successful test, radio switches to the memory test.
Failed test: a new labyrinth is displayed. After three failed tests, the radio is blocked and must be returned to the after -sales service.

- Memory test: six pairs of images appear on the screen ( four rows of three) for 6 seconds and then are hidden. They must be found without making mistakes. (image pairs can be the photos of the models).
_(for the clever ones who would have the idea of ​​taking a photo of the screen, the test would have to start directly after it had been hidden and the images would not be displayed on the screen at the same time at a frequency which would not allow them to be capture only with the camera in long exposure mode.)_

Successful test, radio starts normally.
Failed test, return to the brain test. After three failed tests, the radio is blocked and must be returned to the after -sales service.
